analyses/fusion_filtering/results/pbta-fusion-recurrent-fusion-bysample.tsv

```diff
@@ -71,6 +71,7 @@ BS_2ADZNNV4	0	0	1	0	0
 BS_2AR3AP0N	0	0	0	1	0
 BS_2EMW8ZG3	0	0	0	1	0
 BS_2EN3X6HB	0	0	0	1	0
+BS_2HN96129	0	0	0	1	0
```

Comment:
Doing a quick spot check of these changes - this appears to be due to the random selection of an RNA-seq specimen from samples/participants with multiple RNA-seq specimens. I assume that the selected specimens have changed because `analyses/fusion_filtering/results/pbta-fusion-putative-oncogenic.tsv` changed the ordering of the specimens in some way.

Reply:
thanks for pointing this out, I'll review the changes in the RNA-seq sample selection from this new update.

Reply:
old_sample_selected | new_sample_selected | Kids_First_Participant_ID
---- | ---- | ----
BS_RR12T74P | BS_2HN96129 | PT_DTP4MMRA
BS_5P1TN10Z | BS_TV5B86ZD | PT_9BZETM0M
BS_6VPKXXMR | BS_M85CXHDV | PT_6PP78VNE
BS_7ZV3DPGT | BS_M7QJ8VF0 | PT_QD6KKKJH
BS_5CTVXVRX | BS_8ZS9F31R | PT_DW2XRPC9
BS_8GW1TMV2 | BS_RSRPX3S8 | PT_GR8P8NY5

Since I didn't make changes in `06-recurrent-fusions-per-histology.R` my guess  was maybe order in input file but I can'r recreate the old selection with v16 as well.  I see some of the sample selection changed in #744 as well so not exactly sure how/if we can select the exact same samples each time. 

I looked into set.seed with sample() a bit. 

```
> set.seed(1234)
# sample () selects different in every run if set.seed() is not called exactly before the sample() call
# so it seems to me that the set.seed() at the start of the script doesn't actually apply for all sample() calls in a script?
> sample(LETTERS, 6)
[1] "C" "P" "O" "X" "S" "N"
> sample(LETTERS, 6)
[1] "A" "F" "P" "L" "X" "W"
> sample(LETTERS, 6)
[1] "H" "X" "Z" "T" "G" "F"
```

```
# same selection when set.seed(1234) is run in before each sample() call
> set.seed(1234)
> sample(LETTERS, 6)
[1] "C" "P" "O" "X" "S" "N"
> 
> set.seed(1234)
> sample(LETTERS, 6)
[1] "C" "P" "O" "X" "S" "N"
```

So for these 2 sample() calls in the script I guess I should set.seed() before each of them
https://github.com/kgaonkar6/OpenPBTA-analysis/blob/eacda77df1c372617ba826995e45c6a189dfe39b/analyses/fusion_filtering/06-recurrent-fusions-per-histology.R#L46-L53 

and 

https://github.com/kgaonkar6/OpenPBTA-analysis/blob/eacda77df1c372617ba826995e45c6a189dfe39b/analyses/fusion_filtering/06-recurrent-fusions-per-histology.R#L67-L73

Reply:
Yea, I would expect the behavior above - if you start a new session, set the seed, and sample `LETTERS` 3x, you'd get the same results for all three sequentially, but they wouldn't be identical to one another.

Can you remind me, at a high level, why do we need to do sampling _twice_?  

An idea that comes to mind for keeping things more consistent is sorting the putative oncogenic fusions file by probably a combination of identifiers before doing the sampling (probably upon reading it in).

Reply:
1st Sampling is done in histology file for where there is a matching WGS/WXS for RNA-Seq sample ( that is sample id matches in independentSpecimensFile with has both initial and progreesive/relapse etc )

2nd Sampling  is done  in histology file for samples where we only have RNA-Seq  ( but we need to separate this subset into Initial and Progressive/Relapse since we group for Kids_First_Participant_ID )

Reply:
I ordered the input fusion and clinical files and ran the sampling (3 times and compared each time to the previous one) and that I think does fix the different selection of samples in each run issue. 

This way it'll be release specific selection and we will see random selection changes between release .. right?

Reply:
If you're ordering the file you are sampling by identifiers before sampling and the biospecimens included don't change (and setting a seed), I would _expect_ that you are guarding against changes between releases. Seems like this won't matter much if #795 gets implemented, so I am inclined to get this merged.

Reply:
Thanks for the clarification. .  I'll work on #795